eth/watchers/orchestratorwatcher.go

```diff
@@ -185,27 +186,41 @@ func (ow *OrchestratorWatcher) handleRoundEvent(log types.Log) error {
 		return err
 	}
 
+	wg := &sync.WaitGroup{}
+
 	for _, o := range orchs {
-		if err := ow.cacheOrchestratorStake(ethcommon.HexToAddress(o.EthereumAddr), round); err != nil {
-			glog.Errorf("could not cache stake update for orchestrator %v and round %v", o.EthereumAddr, round)
-		}
+		wg.Add(1)
+		go ow.cacheOrchestratorStake(ethcommon.HexToAddress(o.EthereumAddr), round, wg)
 	}
 
+	wg.Wait()
```

Comment:
I wonder if we need this `WaitGroup` at all.

I think we shouldn't need to wait for all `cacheOrchestratorStake()` to complete before returning from `handleRoundEvent`. The function `handleRoundEvent()` is anyway executed asynchronously and calling `cacheOrchestratorStake()` is the last thing it does.